### PR TITLE
add reverse string exercise

### PR DESCRIPTION
**Pull request by mbtools**

### CODE REVIEW

File: exercises/practice/reverse-string/.meta/zcl_reverse_string.clas.abap [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      IMPORTING
+        input         TYPE string
+      RETURNING 
+        VALUE(result) TYPE string.    
+ENDCLASS.
+
+CLASS zcl_reverse_string IMPLEMENTATION.
+
+  METHOD reverse_string.
+     RAISE EXCEPTION TYPE cx_demo_exception EXPORTING exception_text =
```
Comment (larshp) [inline exercises/practice/reverse-string/.meta/zcl_reverse_string.clas.abap:13]: this exception does not exist in open-abap, suggest replacing with `ASSERT`

File: exercises/practice/reverse-string/zcl_reverse_string.clas.testclasses.abap [hunk 2]
```diff
@@ -0,0 +1,60 @@
+CLASS ltcl_reverse_string DEFINITION FOR TESTING RISK LEVEL HARMLESS DURATION SHORT FINAL.
+
+  PRIVATE SECTION.
+    DATA cut TYPE REF TO zcl_reverse_string.
+    DATA result TYPE string.
```
Comment (larshp) [inline exercises/practice/reverse-string/zcl_reverse_string.clas.testclasses.abap:5]: unused variable?

**Review by larshp — CHANGES_REQUESTED** [review]

**Comment by mbtools** [comment]
I cleaned it up. Assert is not necessary since running the test will fail anyway if it's not implemented properly 😉

File: exercises/practice/reverse-string/.meta/zcl_reverse_string.clas.abap [hunk 3]
```diff
@@ -0,0 +1,17 @@
+CLASS zcl_reverse_string DEFINITION PUBLIC.
+  PUBLIC SECTION.
+    METHOD reverse_string
```
Comment (larshp) [inline exercises/practice/reverse-string/.meta/zcl_reverse_string.clas.abap:3]:
```suggestion
    METHODS reverse_string
```

File: exercises/practice/reverse-string/zcl_reverse_string.clas.abap [hunk 4]
```diff
@@ -0,0 +1,16 @@
+CLASS zcl_reverse_string DEFINITION PUBLIC.
+  PUBLIC SECTION.
+    METHOD reverse_string
```
Comment (larshp) [inline exercises/practice/reverse-string/zcl_reverse_string.clas.abap:3]:
```suggestion
    METHODS reverse_string
```

**Comment by larshp** [comment]
I enabled the linter earlier today, so it runs on both the exercise and solution 😁

**Review by larshp — APPROVED** [review]
👍

**Comment by mbtools** [comment]
note to self... again: do not edit on web

would be good to somehow get this into an abapGit repo. maybe a separate repo that an action could transfer to here

**Pull request merged**